compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -4041,6 +4075,61 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    fn detect_and_explain_multiple_crate_versions(
+        &self,
+        err: &mut Diag<'_>,
+        item_def_id: DefId,
+        item_span: Span,
+        owner: hir::OwnerId,
+        rcvr_ty: Ty<'_>,
+    ) -> bool {
```

Comment:
The bool returned here is very unconventional. Normally, we return `false` if we did not emit something and vice versa. this is a `detect_*` method, so I'd expect the boolean to be flipped. I'd recommend renaming the `suggest` var to something like `trait_in_other_version_found`, and making the return value for this correspond to the convention.